### PR TITLE
#7 Add audience and cleanup_policy fields

### PR DESCRIPTION
**Pull request by SergKam**
Added audience and cleanup_policy fields.
Added validation check to encourage usage of "audience" field.

### CODE REVIEW

File: client/Pages/EventTypeCreate/Models.elm [hunk 1]
```diff
@@ -84,9 +89,12 @@ defaultValues =
     , ( FieldPartitionStrategy, partitionStrategies.random )
     , ( FieldPartitionsNumber, "1" )
     , ( FieldPartitionKeyFields, emptyString )
+    , ( FieldOrderingKeyFields, emptyString )
     , ( FieldRetentionTime, toString defaultRetentionDays )
     , ( FieldSchema, defaultSchema )
     , ( FieldCompatibilityMode, compatibilityModes.forward )
+    , ( FieldAudience, audiences.component_internal )
```
Comment (rcillo) [inline client/Pages/EventTypeCreate/Models.elm]: This field has no default. Does this mean it will be `component_internal` if the user don't change it?

File: client/Pages/EventTypeCreate/Update.elm [hunk 2]
```diff
@@ -310,6 +315,10 @@ submitCreate model =
             , ( "category", asString FieldCategory )
             , ( "partition_strategy", asString FieldPartitionStrategy )
             , ( "partition_key_fields", partitionKeyFields )
+            , ( "ordering_key_fields", orderingKeyFields )
+            , ( "compatibility_mode", asString FieldCompatibilityMode )
```
Comment (rcillo) [inline client/Pages/EventTypeCreate/Update.elm]: Why is it needed to add `compatibility_mode` in this PR?

File: client/Pages/EventTypeCreate/View.elm [hunk 3]
```diff
@@ -172,7 +182,7 @@ viewForm model setup =
                     "Category"
                     ""
                     Help.category
-                    (List.length categoriesOptions == 1)
```
Comment (rcillo) [inline client/Pages/EventTypeCreate/View.elm]: Why is this change needed? Is it because of the order of the fields?

File: client/Pages/EventTypeCreate/Models.elm (hunk 1 above)
Comment (SergKam) [reply client/Pages/EventTypeCreate/Models.elm]: Yes, exactly, and I don't see any problem with it. This kind of decision should not stop user fro what he wants to do right now "I want to create ET in two clicks". This field can be changed any time later. So I choose the safest option "internal" only, so later you can expose it gradually to a broader and broader audience.

File: client/Pages/EventTypeCreate/View.elm (hunk 3 above)
Comment (SergKam) [reply client/Pages/EventTypeCreate/View.elm]: It was moved to the select components https://github.com/zalando-incubator/nakadi-ui/blob/a614a5fa3cc18d490de26fd5905b930d9f1bbb8f/client/Pages/EventTypeCreate/View.elm#L387 because it is used many times this way.

File: client/Pages/EventTypeDetails/Help.elm [hunk 4]
```diff
@@ -387,3 +387,71 @@ subscription =
...
+    [ text "Event type cleanup policy."
+    , newline
+    , bold "Possible cleanup policies are:"
+    , newline
+    , text "- "
+    , mono "delete"
+    , text " will delete old events after retention time expires."
+    , newline
+    , text "- "
+    , mono "compact"
```
Comment (rcillo) [inline client/Pages/EventTypeDetails/Help.elm]: @SergKam could you please add also the following text in a very visible way:

"Log compacted event types **must not**  contain personal identifiable information in accordance to GDPR. If you plan to store user data permanently, check the legal department of your organization".

File: client/Pages/EventTypeCreate/Update.elm (hunk 2 above)
Comment (SergKam) [reply client/Pages/EventTypeCreate/Update.elm]: good catch! some merge bug

File: client/Pages/EventTypeDetails/Help.elm (hunk 4 above)
Comment (SergKam) [reply client/Pages/EventTypeDetails/Help.elm]: screenshot 35

File: client/Pages/EventTypeCreate/Models.elm (hunk 1 above)
Comment (rcillo) [reply client/Pages/EventTypeCreate/Models.elm]: 👍

File: client/Pages/EventTypeDetails/Help.elm (hunk 4 above)
Comment (rcillo) [reply client/Pages/EventTypeDetails/Help.elm]: @SergKam thank you. My mind also refuses to accept this, but it's when it comes to legal, we developer have a weaker hand on negotiations and discussions. If the guy on the other side has the authority in terms of security/legal, what can we do right? But I hope we can change it in the future once we define an API for deletion of events and the data classification process gets fully automated.

**Comment by rcillo** [comment]
👍
